src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanFL401Service.java

```diff
@@ -1,42 +1,169 @@
+
+        FormType formType = getCaseType();
+
+        Map<String, String> inputFieldsMap = getOcrDataFieldAsMap(inputFieldsList);
+
+        List<String> unknownFieldsList = null;
+
+        BulkScanFormValidationConfigManager.ValidationConfig validationConfig =
+                configManager.getValidationConfig(formType);
+
+        Map<String, Object> populatedMap =
+                (Map<String, Object>)
+                        BulkScanTransformHelper.transformToCaseData(
+                                new HashMap<>(
+                                        transformConfigManager
+                                                .getTransformationConfig(formType)
+                                                .getCaseDataFields()),
+                                inputFieldsMap);
+
+        populatedMap.put(SCAN_DOCUMENTS, transformScanDocuments(bulkScanTransformationRequest));
+
+        Map<String, String> caseTypeAndEventId =
+                transformConfigManager.getTransformationConfig(formType).getCaseFields();
+
+        bulkScanFL401ConditionalTransformerService.transform(
+                populatedMap, inputFieldsMap, bulkScanTransformationRequest);
+
+        BulkScanTransformationResponse.BulkScanTransformationResponseBuilder builder =
+                BulkScanTransformationResponse.builder()
+                        .caseCreationDetails(
+                                CaseCreationDetails.builder()
+                                        .caseTypeId(caseTypeAndEventId.get(CASE_TYPE_ID))
+                                        .eventId(caseTypeAndEventId.get(EVENT_ID))
+                                        .caseData(populatedMap)
+                                        .build());
+
+        if (nonNull(validationConfig)) {
+            unknownFieldsList =
+                    bulkScanValidationHelper.findUnknownFields(
+                            inputFieldsList,
+                            validationConfig.getMandatoryFields(),
+                            validationConfig.getOptionalFields());
+        }
+        BulkScanGroupValidatorUtil.updateTransformationUnknownFieldsByGroupFields(
+                formType, unknownFieldsList, builder);
+        return builder.build();
+    }
+
+    private Map<String, String> getOcrDataFieldsMap(List<OcrDataField> ocrDataFields) {
+        return null != ocrDataFields
+                ? ocrDataFields.stream()
+                        .collect(Collectors.toMap(OcrDataField::getName, OcrDataField::getValue))
+                : null;
+    }
+
+    private List<String> validateInputDate(
+            List<OcrDataField> ocrDataFields, String fieldName, String message) {
+
+        final Map<String, String> ocrDataFieldsMap = getOcrDataFieldsMap(ocrDataFields);
+
+        if (null != ocrDataFieldsMap
+                && ocrDataFieldsMap.containsKey(fieldName)
+                && hasText(ocrDataFieldsMap.get(fieldName))) {
+            String date = ocrDataFieldsMap.get(fieldName);
```

Comment:
Could you change the date variable name to something more meaningful like "dateInput", "dateField" or "dateStr"